pkg/controllers/dynakube/extension/service.go

```diff
@@ -0,0 +1,92 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/service"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (r *reconciler) reconcileService(ctx context.Context) error {
+	if r.dk.PrometheusEnabled() {
+		return r.createOrUpdateService(ctx)
+	} else {
```

Comment:
is the `else` necessary?

```go
if r.dk.PrometheusEnabled() {
	return r.createOrUpdateService(ctx)
}

// do clean-up
if meta.FindStatusCondition(*r.dk.Conditions(), extensionsServiceConditionType) == nil {
	return nil
}
defer meta.RemoveStatusCondition(r.dk.Conditions(), extensionsServiceConditionType)
....
```
wouldn't this do the same?

Reply:
the pattern in the other reconcilers tends to be:
```go
if !enabled {
 	// do clean-up
	return nil
}

// do reconcile
return nil
```

I do not really have a preference either way, just noting it 😉

Reply:
fixed in d19559f5